ftplugin/python/jedi.vim

```diff
@@ -44,10 +44,15 @@ if g:jedi#auto_initialization
         inoremap <silent> <buffer> <space> <C-R>=jedi#smart_auto_mappings()<CR>
     end
 
-    if g:jedi#auto_close_doc
+    if g:jedi#auto_close_doc && (&g:completeopt =~# '\<preview\>' && &g:completeopt !~# '\<popup\>')
         " close preview if its still open after insert
         augroup jedi_preview
-            autocmd! InsertLeave <buffer> if pumvisible() == 0|pclose|endif
+            if v:version > 704
```

Comment:
What's the reasoning for this change?

Reply:
Hello, 

I just implemented [the first two answers](https://stackoverflow.com/questions/3105307/how-do-you-automatically-remove-the-preview-window-after-autocompletion-in-vim) as `:help g:jedi#auto_close_doc` states that

> this window is being closed after insertion of a completion item.

the `CompleteDone` event seemed the more exact autocmd to achieve this, whenever available.

Reply:
Thanks that makes sense.